File: R/demos/demGpSample.R
## DEMGPSAMPLE Simple demonstration of sampling from a covariance function.
demGpSample <- function( bw=FALSE ) {
  require(fields)

  x = as.matrix(seq(-1,1,length=25))
  kern = kernCreate(x, 'rbf')
  kern$inverseWidth = 10

  graphics.off() ## kill all devices
  par(pty="s") #figure(1)
  par(mar=c(4, 4, 2, 6)) ## increase right margin for legend
  K = kernCompute(kern, x)
  colormap=heat.colors(256)
  if (bw)
    colormap=gray.colors(256)

  image(1:dim(K)[2], 1:dim(K)[1] , K[dim(K)[2]:1, ], xlab = "n", ylab = "n")
  image.plot(legend.only = TRUE, nlevel = 256, zlim = c(min(K),max(K)), col = colormap) ## Colorbar
  dev.copy2eps(file = "~/mlprojects/gp/R/inst/doc/gpCovariance.eps") ## Save plot as eps
  ## Convert to png. Needs the 'eps2png' facility. If not already installed: 'sudo apt-get install eps2png'
  system('eps2png ~/mlprojects/gp/R/inst/doc/gpCovariance.eps')

#   if (exists('printDiagram') && exists('var') && exists(printDiagram))
#     printPlot('gpCovariance', '../tex/diagrams/', '../html') ## !!!

  ## Need to take the real part of the sample, as the kernel is numerically less than full rank.
  f = t(Re(gaussSamp(K, 1)))

  dev.new() #figure(2)
  a = plot(f, xlab='n', ylab= bquote(f[n]))
  dev.copy2eps(file = "~/mlprojects/gp/R/inst/doc/gpSample.eps") ## Save plot as eps
  ## Convert to png. Needs the 'eps2png' facility. If not already installed: 'sudo apt-get install eps2png'
  system('eps2png ~/mlprojects/gp/R/inst/doc/gpSample.eps')

#   if (exists('printDiagram') && exists('var') && exists(printDiagram))
#     printPlot('gpSample', '../inst/doc/', '../html') ## !!!

  # save(K, x, f, file='demGpSample.RData')
  return (list(K=K, f=f, x=x))
}

File: gpmgp/demGpSample.m
% DEMGPSAMPLE Simple demonstration of sampling from a covariance function.

% GP

randn('seed', 1e5)
rand('seed', 1e5)

bw = false;
x = linspace(-1, 1, 25)';
kern = kernCreate(x, 'rbf');
kern.inverseWidth = 10;

figure(1)
clf
K = kernCompute(kern, x);
imagesc(K);
if bw
  colormap gray
else
  colormap jet
end  
colorbar
t = [];
t = [t xlabel('n')];
t = [t ylabel('m')];
set(gca, 'fontname', 'times')
set(gca, 'fontsize', 24)
if exist('printDiagram', 'var') && printDiagram
  printPlot('gpCovariance', '../tex/diagrams/', '../html');
end
% need to take the real part of the sample as the kernel is numerically less than full rank 
f = real(gsamp(zeros(1, size(x, 1)), K, 1))';

figure(2) 
clf
a = plot(f, 'k.');
t = [t text(12.5, -.5, 'n')];
t = [t text(-5, .5, 'f_n')];
set(t, 'fontname', 'times')
set(t, 'fontsize', 24)
set(t, 'fontangle', 'italic')
set(gca, 'fontname', 'times')
set(gca, 'fontsize', 18)
set(a,'markersize', 20)
set(a, 'linewidth', 2)
zeroAxes(gca)
if exist('printDiagram', 'var') && printDiagram
  printPlot('gpSample', '../tex/diagrams/', '../html');
end
% if bw
%   set(a, 'color', [0 0 0])
%   print('-deps', '../tex/diagrams/gpSamplebw.eps');
% else
%   print('-depsc', '../tex/diagrams/gpSample.eps');
% end

save demGpSample K x f
